### PR TITLE
Add dbTokenClientID to RP Config

### PR DESCRIPTION
**Pull request by bennerv**
### Which issue this PR addresses:

Fixes https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/10378550

### What this PR does / why we need it:

Public cloud context.  Adapt URLs for gov or other cloud

In https://github.com/Azure/ARO-RP/pull/1550 we switched from a hardcoded sts.windows.net as a token issuer to a dynamic cloud AAD URL (public cloud: login.microsoft.net).  The v1 OIDC token that was being requested always sets the issuer to sts.windows.net in pub cloud which causes verification issues when accepting a new token. 

As a result, https://github.com/Azure/ARO-RP/pull/1611 was updated in order to use the v2 endpoint which sets the issuer claim properly to the token endpoints shown in the app registration.  But the app registration itself still returns a v1 token when requested unless we set the     "accessTokenAcceptedVersion": 2 on the app registration. [1]

If we use v2 on the app registration, it no longer uses the redirect URL for the `aud` claim, and instead uses the `client_id` of the app registration, hence this change.  [2]

References:
[1] https://docs.microsoft.com/en-us/azure/active-directory/develop/reference-app-manifest#accesstokenacceptedversion-attribute
[2] https://docs.microsoft.com/en-us/azure/active-directory/develop/access-tokens#payload-claims

### Test plan for issue:
Unit tests, e2e, and deploying the rp in dev 

### Is there any documentation that needs to be updated for this PR?

RP config has been updated.  
deploy-shared-environment has been updated.

### CODE REVIEW

File: pkg/deploy/generator/resources_rp.go [hunk 1]
```diff
@@ -376,34 +376,35 @@ func (g *generator) rpVMSS() *arm.Resource {
...
+		"armClientId",
+		"azureCloudName",
 		"billingE2EStorageAccountId",
 		"clusterMdmAccount",
 		"clusterMdsdAccount",
 		"clusterMdsdConfigVersion",
 		"clusterMdsdNamespace",
 		"clusterParentDomainName",
+		"databaseAccountName",
+		"dbtokenClientId",
```
Comment (nilsanderselde) [inline pkg/deploy/generator/resources_rp.go:391]: Would it make sense to split this into two commits, the first one just reordering these strings, and the second one adding the new value (and making the other changes)?

Comment (bennerv) [reply pkg/deploy/generator/resources_rp.go:391]: Seems pretty small to split into it's own PR as it's only this reordering.  I can put it in another commit in this PR if that makes more sense.

Comment (nilsanderselde) [reply pkg/deploy/generator/resources_rp.go:391]: Yeah, two commits in the same PR is what I was suggesting.

Comment (bennerv) [reply pkg/deploy/generator/resources_rp.go:391]: Misread it my bad :+1:   Will split.

Comment (bennerv) [reply pkg/deploy/generator/resources_rp.go]: Done

**Review by nilsanderselde — APPROVED** [review]

File: docs/prepare-a-shared-rp-development-environment.md [hunk 2]
```diff
@@ -228,6 +228,24 @@ locations.
...
+
+   ```bash
+   AZURE_DBTOKEN_CLIENT_ID="$(az ad app create --display-name dbtoken \
+      --oauth2-allow-implicit-flow false \
+      --query appId \
+      -o tsv)"
+
+   OBJ_ID="$(az ad app show --id $AZURE_DBTOKEN_CLIENT_ID --query objectId)"
+
+   # TODO - currently the az-cli cannot update the requestedAccessTokenVersion
```
Comment (m1kola) [inline docs/prepare-a-shared-rp-development-environment.md:243]: I would remove this TODO: a) not clear what one needs to do here and b) `az rest` is ok.

P.S.: `az rest` is part of az cli :)

File: deploy/rp-production-parameters.json [hunk 3]
```diff
@@ -47,6 +47,9 @@
         "databaseAccountName": {
             "value": ""
         },
+        "dbtokenClientId": {
```
Comment (m1kola) [inline deploy/rp-production-parameters.json:50]: Do we need to update the prod config to include a value for this?

**Review by m1kola — COMMENTED** [review]
One questions.

File: deploy/rp-production-parameters.json (hunk 3 above)
Comment (bennerv) [reply deploy/rp-production-parameters.json:50]: Yes.  I need elevated JIT when this is merged in to create these.

**Review by bryanro92 — DISMISSED** [review]
lgtm

**Review by troy0820 — COMMENTED** [review]
LGTM

**Review by mjudeikis — APPROVED** [review]

**Pull request merged**